code/tests/component/movement/test_ShootsEntity.gd

```diff
@@ -0,0 +1,72 @@
+extends "res://addons/gut/test.gd"
+
+var component_cls = preload('res://src/component/ShootsEntity.tscn')
+
+var comp
+
+func setup():
+	comp = component_cls.instance()
+	add_child(comp)
+
+func teardown():
+	remove_child(comp)
+
+func test_init():
+	assert_true(comp != null)
+
+func test_process_counts_seconds_since_last_shot():
+	comp.seconds_since_last_shot = 0
+	comp.bullets_per_second = 0.05
+	var delta = 499
+
+	comp._process(delta)
+
+	assert_eq(comp.seconds_since_last_shot, delta)
+
+func test_process_calls_shoot_if_not_cooldown():
+	comp.seconds_since_last_shot = 999
+
+	comp._process(0)
+
+	# i.e: shoot was called
+	assert_eq(comp.seconds_since_last_shot, 0)
+
+func test_process_doesnt_call_if_cooldown():
+	Input.action_press('shoot')
+	comp.seconds_since_last_shot = -1
+
+	comp._process(0)
+
+	assert_eq(comp.seconds_since_last_shot, -1)
+
+func test_create_bullet_creates_and_prepares_bullet():
+	comp.global_position = Vector2()
+	comp.bullet_speed = 1500
+
+	var error_margin = 0.05
+
+	var test_table = { # mouse position: velocity after rotation
+		Vector2(-194, 17): Vector2(-1494.273804, 130.941422),
```

Comment:
This is pretty fragile, but I don't see how/what else we can test here.

Reply:
How is this fragile? We control a few variables and freeze all others, and expect a static output. I think that's pretty solid.

Reply:
Floating-point numbers is one of those common sources of problems due to rounding, imprecision, etc. I can easily imagine an updated version of Godot differs by 0.00001 for one of these values and the test fails, or you tweak gravity and the test fails, etc.

Anyway, it's fine, if it starts flaking out we can always tweak it.

Reply:
Notice that we use `assert_almost_eq` instead of `assert_eq` wi